R-package/R/callback.R

```diff
@@ -37,7 +37,12 @@ cb.reset.parameters <- function(new_params) {
 
     # Some parameters are not allowed to be changed,
     # since changing them would simply wreck some chaos
-    not_allowed <- c("num_class", "metric", "boosting_type")
+    not_allowed <- c(
+      "num_class"
+      , "metric"
+      , "boosting"
```

Comment:
Refer to https://github.com/microsoft/LightGBM/pull/2443#discussion_r329503844.

Reply:
changed, thank you! Now it is

```r
not_allowed <- c(
      .PARAMETER_ALIASES()[["num_class"]]
      , .PARAMETER_ALIASES()[["metric"]]
      , .PARAMETER_ALIASES()[["boosting"]]
    )
```

Reply:
I think it's much better now! Thanks!